lib/parse.js

```diff
@@ -61,7 +69,10 @@ module.exports = function parse(html, options) {
         }
 
         if (!isOpen || current.voidElement) {
-            level--;
+            if (level > -1 && (current.voidElement || current.name === tag.slice(2, tag.indexOf(' ')))) {
```

Comment:
I'm a little confused about what `tag.slice(2, tag.indexOf(' ')))` is doing-- is looking for a space for the second slice index appropriate?